### PR TITLE
47295 RNW show selection

### PR DESCRIPTION
**Pull request by ganpat-tech**
PR contain.
1)Code changes
2)Unit test cases
3)Function test cases

### CODE REVIEW

File: Tests/pageObject/RadioNavigationalWarningsAdminListEndUser.page.ts [hunk 1]
```diff
@@ -102,4 +119,27 @@ export default class RadioNavigationalWarningsListEndUser {
     }
     expect(compareDate).toBeTruthy();
   }
+
+  public async verifySelectOption() {
```
Comment (nevillejrbrown) [inline Tests/pageObject/RadioNavigationalWarningsAdminListEndUser.page.ts:123]: This method appears to be doing a series of actions and checks.  Can you split this up?

**Review by nevillejrbrown — CHANGES_REQUESTED** [review]

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWService.cs [hunk 2]
```diff
@@ -199,5 +199,23 @@ public async Task<bool> EditRadioNavigationalWarningsRecord(EditRadioNavigationa
             return true;
         }
 
+        public async Task<List<RadioNavigationalWarningsData>> ShowRadioNavigationalWarningsData(string correlationId, int[] data)
+        {
+            try
+            {
+                _logger.LogInformation(EventIds.RNWShowListDetailFromDatabaseStarted.ToEventId(), "Maritime safety information request to show RNW details from database started for _X-Correlation-ID:{correlationId}", correlationId);
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWService.cs:206]:
```suggestion
                _logger.LogInformation(EventIds.RNWShowListDetailFromDatabaseStarted.ToEventId(), "Maritime safety information request to show RNW details for selected warnings from database started for _X-Correlation-ID:{correlationId}", correlationId);
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWService.cs [hunk 3]
```diff
@@ -199,5 +199,23 @@ public async Task<bool> EditRadioNavigationalWarningsRecord(EditRadioNavigationa
...
+
+                List<RadioNavigationalWarningsData> radioNavigationalWarningsData = await _rnwRepository.ShowRadioNavigationalWarningsDataList(data);
+
+                _logger.LogInformation(EventIds.RNWShowListDetailFromDatabaseCompleted.ToEventId(), "Maritime safety information request to show RNW details from database completed for _X-Correlation-ID:{correlationId}", correlationId);
+
+                return radioNavigationalWarningsData;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(EventIds.ErrorInRNWShowListDetailFromDatabase.ToEventId(), ex, "Maritime safety information error has occurred in the process to show RNW detail from database with Exception:{ex} and _X-Correlation-ID:{correlationId}", ex.Message, correlationId);
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWService.cs:216]:
```suggestion
                _logger.LogError(EventIds.ErrorInRNWShowListDetailFromDatabase.ToEventId(), ex, "Maritime safety information error has occurred in the process to show RNW details for selected warnings from database with Exception:{ex} and _X-Correlation-ID:{correlationId}", ex.Message, correlationId);
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js [hunk 4]
```diff
@@ -31,3 +31,47 @@ function RnwFilterList(pageSelectedIndex) {
     rnwUrl = rnwUrl.replace("amp;", "");
     window.location.href = rnwUrl;
 }
+
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js:34]: As we are splitting Admin Web App, please add separate js file for public RNW pages.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js [hunk 5]
```diff
@@ -31,3 +31,47 @@ function RnwFilterList(pageSelectedIndex) {
...
+function do_Selection() {
+    var checkboxes = document.getElementsByName('checkbox');
+    var button = document.getElementById('select_button');
+
+    if (button.value == 'Select all') {
+        for (var i in checkboxes) {
+            checkboxes[i].checked = 'FALSE';
+            document.getElementById("BtnShowSelection").disabled = false;
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js:42]: document.getElementById("BtnShowSelection").disabled = false;
This statement should be outside of for loop.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js [hunk 6]
```diff
@@ -31,3 +31,47 @@ function RnwFilterList(pageSelectedIndex) {
...
+        }
+        button.value = 'Clear all'
+    } else {
+        for (var i in checkboxes) {
+            checkboxes[i].checked = '';
+            document.getElementById("BtnShowSelection").disabled = true;
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js:48]: document.getElementById("BtnShowSelection").disabled = true;
This statement should be outside of for loop.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js [hunk 7]
```diff
@@ -31,3 +31,47 @@ function RnwFilterList(pageSelectedIndex) {
...
+        }
+        button.value = 'Select all';
+    }
+}
+
+function showSelection_() {
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js:54]:
```suggestion
function showSelection() {
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js [hunk 8]
```diff
@@ -31,3 +31,47 @@ function RnwFilterList(pageSelectedIndex) {
...
+    var checkboxes = document.getElementsByName('checkbox');
+    var len_ = checkboxes.length;
+    var arr = [];
+
+    for (var i in checkboxes) {
+        if (checkboxes[i].checked && i < len_) {
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js:60]: Can directly use checkboxes.length and remove variable len_ .
```suggestion
        if (checkboxes[i].checked && i < checkboxes.length) {
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js [hunk 9]
```diff
@@ -31,3 +31,47 @@ function RnwFilterList(pageSelectedIndex) {
...
+            arr.push(((document.getElementById("Id_" + i)).innerHTML).trim());
+        }
+    }
+    document.getElementById('showSelectionId').value = arr
+}
+
+function terms_changed() {
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js:67]: Better name can be enableDisableShowSelection()

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js [hunk 10]
```diff
@@ -31,3 +31,47 @@ function RnwFilterList(pageSelectedIndex) {
...
+    var checkboxes = document.getElementsByName('checkbox');
+    var len_ = checkboxes.length;
+    document.getElementById("BtnShowSelection").disabled = true;
+
+    for (var i in checkboxes) {
+        if (checkboxes[i].checked && i < len_) {
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js:73]: Same as above, can directly use checkboxes.length and remove variable len_ .
```suggestion
        if (checkboxes[i].checked && i < checkboxes.length) {
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js [hunk 11]
```diff
@@ -31,3 +31,47 @@ function RnwFilterList(pageSelectedIndex) {
...
+            document.getElementById("BtnShowSelection").disabled = true;
+        }
+        button.value = 'Select all';
+    }
+}
+
+function showSelection_() {
+    var checkboxes = document.getElementsByName('checkbox');
+    var len_ = checkboxes.length;
+    var arr = [];
```
Comment (usha13529) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js:57]:
```suggestion
    var selectedIds= [];
```

**Review by usha13529 — CHANGES_REQUESTED** [review]
Few comments & suggestions to look into please.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/RadioNavigationalWarningsController.cs [hunk 12]
```diff
@@ -38,5 +38,22 @@ public async Task<IActionResult>  About()
 
             return View();
         }
+
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/RadioNavigationalWarningsController.cs:41]: Thanks for highlighting ,added integration test case.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/RadioNavigationalWarningsController.cs [hunk 13]
```diff
@@ -38,5 +38,22 @@ public async Task<IActionResult>  About()
...
+        [HttpPost]
+        public async Task<IActionResult> ShowSelection()
+        {
+            int[] nums = Array.Empty<int>();
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/RadioNavigationalWarningsController.cs]: Thanks for highlighting, done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/Interfaces/IRNWRepository.cs [hunk 14]
```diff
@@ -13,5 +13,6 @@ public interface IRNWRepository
         Task<DateTime> GetRadioNavigationalWarningsLastModifiedDateTime();
         EditRadioNavigationalWarningAdmin GetRadioNavigationalWarningById(int id);
         Task UpdateRadioNavigationalWarning(EditRadioNavigationalWarningAdmin radioNavigationalWarning);
+        Task<List<RadioNavigationalWarningsData>> ShowRadioNavigationalWarningsDataList(int[] data);
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/Interfaces/IRNWRepository.cs]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/Interfaces/IRNWService.cs [hunk 15]
```diff
@@ -12,5 +12,6 @@ public interface IRNWService
         Task<string> GetRadioNavigationalWarningsLastModifiedDateTime(string correlationId);
         EditRadioNavigationalWarningAdmin GetRadioNavigationalWarningById(int id, string correlationId);
         Task<bool> EditRadioNavigationalWarningsRecord(EditRadioNavigationalWarningAdmin radioNavigationalWarning, string correlationId);
+        Task<List<RadioNavigationalWarningsData>> ShowRadioNavigationalWarningsData(string correlationId, int[] data);
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/Interfaces/IRNWService.cs]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWService.cs [hunk 16]
```diff
@@ -199,5 +199,23 @@ public async Task<bool> EditRadioNavigationalWarningsRecord(EditRadioNavigationa
...
 
+        public async Task<List<RadioNavigationalWarningsData>> ShowRadioNavigationalWarningsData(string correlationId, int[] data)
+        {
+            try
+            {
+                _logger.LogInformation(EventIds.RNWShowListDetailFromDatabaseStarted.ToEventId(), "Maritime safety information request to show RNW details from database started for _X-Correlation-ID:{correlationId}", correlationId);
+
+                List<RadioNavigationalWarningsData> radioNavigationalWarningsData = await _rnwRepository.ShowRadioNavigationalWarningsDataList(data);
+
+                _logger.LogInformation(EventIds.RNWShowListDetailFromDatabaseCompleted.ToEventId(), "Maritime safety information request to show RNW details from database completed for _X-Correlation-ID:{correlationId}", correlationId);
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWService.cs]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/RadioNavigationalWarningsController.cs [hunk 17]
```diff
@@ -38,5 +38,22 @@ public async Task<IActionResult>  About()
...
+            string data = Request.Form["showSelectionId"];
+            if (!string.IsNullOrWhiteSpace(data))
+            {
+                nums = data.Split(',').Select(n => Convert.ToInt32(n)).ToArray();
+            }
+
+            ViewBag.LastModifiedDateTime = await _rnwService.GetRadioNavigationalWarningsLastModifiedDateTime(GetCurrentCorrelationId());
+
+            List<RadioNavigationalWarningsData> radioNavigationalWarningsData = await _rnwService.ShowRadioNavigationalWarningsData(GetCurrentCorrelationId(), nums);
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/RadioNavigationalWarningsController.cs]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWService.cs (hunk 2 above)
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWService.cs]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWService.cs (hunk 3 above)
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/RNWService.cs]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js (hunk 5 above)
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js (hunk 6 above)
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js (hunk 7 above)
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js (hunk 8 above)
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js (hunk 9 above)
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js (hunk 10 above)
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js (hunk 11 above)
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js (hunk 4 above)
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/wwwroot/js/Custom/RadioNavigationalWarnings-admin.js]: done required changes

File: Tests/pageObject/RadioNavigationalWarningsAdminListEndUser.page.ts (hunk 1 above)
Comment (jigar15426) [reply Tests/pageObject/RadioNavigationalWarningsAdminListEndUser.page.ts]: resolved

**Review by usha13529 — APPROVED** [review]
Thanks for the changes @ganpat-tech @jigar15426. Looks good to me.

**Review by barrie-cooper — COMMENTED** [review]
Just minor naming suggestions

**Review by nevillejrbrown — APPROVED** [review]

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/Interfaces/IRNWRepository.cs [hunk 18]
```diff
@@ -13,5 +13,6 @@ public interface IRNWRepository
         Task<DateTime> GetRadioNavigationalWarningsLastModifiedDateTime();
         EditRadioNavigationalWarningAdmin GetRadioNavigationalWarningById(int id);
         Task UpdateRadioNavigationalWarning(EditRadioNavigationalWarningAdmin radioNavigationalWarning);
+        Task<List<RadioNavigationalWarningsData>> ShowRadioNavigationalWarningsDataList(int[] selectedIds);
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/Interfaces/IRNWRepository.cs]: Thanks For Highlighting, Done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/Interfaces/IRNWService.cs [hunk 19]
```diff
@@ -12,5 +12,6 @@ public interface IRNWService
         Task<string> GetRadioNavigationalWarningsLastModifiedDateTime(string correlationId);
         EditRadioNavigationalWarningAdmin GetRadioNavigationalWarningById(int id, string correlationId);
         Task<bool> EditRadioNavigationalWarningsRecord(EditRadioNavigationalWarningAdmin radioNavigationalWarning, string correlationId);
+        Task<List<RadioNavigationalWarningsData>> ShowRadioNavigationalWarningsData(int[] selectedIds, string correlationId);
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/Interfaces/IRNWService.cs]: Done required changes

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWRepositoryTest.cs [hunk 20]
```diff
@@ -127,6 +127,14 @@ public async Task WhenCallGetRadioNavigationalWarningsDataList_ThenReturnOnlyNon
             Assert.AreEqual(2, result.Count);
         }
 
+        [Test]
+        public async Task WhenCallShowRadioNavigationalWarningsDataList_ThenReturnOnlyNonDeletedAndNonExpiredWarnings()
```
Comment (barrie-cooper) [inline Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWRepositoryTest.cs:131]:
```suggestion
        public async Task WhenCallGetSelectedRadioNavigationalWarningsDataList_ThenReturnOnlyNonDeletedAndNonExpiredWarnings()
```

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWServiceTest.cs [hunk 21]
```diff
@@ -224,6 +224,23 @@ public void WhenCallGetRadioNavigationalWarningsDataListWithException_ThenReturn
                                async delegate { await _rnwService.GetRadioNavigationalWarningsData(string.Empty); });
         }
 
+        [Test]
+        public async Task WhenCallShowRadioNavigationalWarningsDataList_ThenReturnWarnings()
```
Comment (barrie-cooper) [inline Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWServiceTest.cs:228]:
```suggestion
        public async Task WhenCallGetSelectedRadioNavigationalWarningsDataList_ThenReturnWarnings()
```

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWServiceTest.cs [hunk 22]
```diff
@@ -224,6 +224,23 @@ public void WhenCallGetRadioNavigationalWarningsDataListWithException_ThenReturn
...
+        {
+            A.CallTo(() => _fakeRnwRepository.GetSelectedRadioNavigationalWarningsDataList(Array.Empty<int>())).Returns(GetFakeRadioNavigationalWarningsDataList());
+            List<RadioNavigationalWarningsData> result = await _rnwService.GetSelectedRadioNavigationalWarningsData(Array.Empty<int>(), string.Empty);
+            Assert.AreEqual(1, result.Count);
+        }
+
+        [Test]
+        public void WhenCallShowRadioNavigationalWarningsDataListWithException_ThenReturnException()
```
Comment (barrie-cooper) [inline Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWServiceTest.cs:236]:
```suggestion
        public void WhenCallGetSelectedRadioNavigationalWarningsDataListWithException_ThenReturnException()
```

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarnings/RadioNavigationalWarningsTest.cs [hunk 23]
```diff
@@ -60,6 +62,28 @@ public async Task WhenCallAbout_ThenReturnLastModifiedDateTime()
             Assert.AreEqual("150215 UTC Aug 19", lastModifiedDateTime);
         }
 
+        [Test]
+        public async Task WhenCallShowRadioNavigationalWarningsDataList_ThenReturnOnlyNonDeletedAndNonExpiredWarnings()
```
Comment (barrie-cooper) [inline UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarnings/RadioNavigationalWarningsTest.cs:66]:
```suggestion
        public async Task WhenCallGetSelectedRadioNavigationalWarningsDataList_ThenReturnOnlyNonDeletedAndNonExpiredWarnings()
```

**Review by barrie-cooper — COMMENTED** [review]
Please update the test names to reflect the changed method names

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWRepositoryTest.cs (hunk 20 above)
Comment (ganpat-tech) [reply Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWRepositoryTest.cs]: Thanks for highlighting, done required changes

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWServiceTest.cs (hunk 21 above)
Comment (ganpat-tech) [reply Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWServiceTest.cs]: done required changes

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWServiceTest.cs (hunk 22 above)
Comment (ganpat-tech) [reply Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/RNWServiceTest.cs]: done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarnings/RadioNavigationalWarningsTest.cs (hunk 23 above)
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarnings/RadioNavigationalWarningsTest.cs]: done required changes

**Pull request merged**